File: 2017-08-07-326-easy-nearest-prime-numbers/main.cpp
#include <iostream>

bool is_prime(long long n)
{
	for (long long i = 2; i < std::sqrt(n); i++) {
		if (n % i == 0) return true;
	}
	return false;
}

int main()
{
	
	return 0;
}

File: 2017-08-07-326-easy-nearest-prime-numbers/main.swift
import Foundation

func isPrime(_ n: Int) -> Bool {
    guard n % 2 == 1 else { return false }
    guard n % 3 != 0 else { return false }
    let squareRoot = Int(sqrt(Double(n)))
    for i in 5...squareRoot {
        if n % i == 0 { return false }
    }
    return true
}

func findNearestPrime(for n: Int) -> String {
    guard !isPrime(n) else { return "\(n) is prime." }
    var down = n, up = n
    while !isPrime(down) { down -= 1 }
    while !isPrime(up) { up += 1 }
    return "\(down) < \(n) < \(up)"
}
